doc/man3/EVP_PKEY_check.pod

```diff
@@ -18,7 +18,7 @@ EVP_PKEY_private_check, EVP_PKEY_pairwise_check
 
 =head1 DESCRIPTION
 
-EVP_PKEY_param_check() validates the domain parameters component of the key
+EVP_PKEY_param_check() validates the parameters component of the key
```

Comment:
I would probably leave the domain in this. That should be what is getting validated - not the other parameters

Reply:
I'm thinking that it's up to the provider to decide exactly what gets validated.  For *our* stuff, it absolutely makes sense to limit that to domain parameters, but other stuff we haven't heard about may have other parameters that aren't necessarily *domain* they want to include in the validation.
(that is, if everyone even agree on what constitutes *domain* parameters...  recent discussions suggest that is not always the case.  Me, I'm actually rather confused on the matter)

Reply:
The whole point of validation is that is comes from specs (particularly the SP800-56A/B keyagreement spec's - which is all about assurance that keys are valid. i.e. validation of domain params/ pub/priv and keypair checks. I dont see there being these other types of checks.. If there are then we add them when required. The other_parameter's does not fit into this scheme at all.

Reply:
This falls under the category of "know your provider", methinks, and also "know your algorithm"...

... which also makes me think that we're being sloppy regarding documentation.  Just like we have `EVP_MAC-something.pod` and `EVP_KDF-whatever.pod`, we should have a few `EVP_KEYMGMT-name.pod` and similar, where we describe all sorts of things that are specific to our implementation of the different key algorithms / types.

(and yeah, there should be `EVP_SIGNATURE-algo.pod` too, and...)